src/command/upload.ts

```diff
@@ -82,10 +94,17 @@ export class Upload extends RootCommand implements LeafCommand {
 
     this.console.dim('Waiting for file chunks to be synced on Swarm network...')
     //refresh tag before populate tracking
-    tag = await this.bee.retrieveTag(tag.uid)
-    const synced = await this.waitForFileSynced(tag)
 
-    if (!synced) return //error message printed before
+    if (this.skipSync) {
+      this.console.info('Skipping synchronization')
+    } else if (tag) {
+      tag = await this.bee.retrieveTag(tag.uid)
+      const synced = await this.waitForFileSynced(tag)
+
+      if (!synced) return //error message printed before
+    } else {
+      this.console.error('Tag was not created')
+    }
```

Comment:
Could it happen theoritically? I mean `tag = await this.bee.createTag()` would throw an error if this happens, doesn't it?

Reply:
You are right.

I originally added it for better readability since it is not very obvious that when `this.skipSync` is turned off, then `tag` always exists, but it does not add much value.